include/fastdds-statistics-backend/types/EntityId.hpp

```diff
@@ -19,31 +19,155 @@
 #ifndef _EPROSIMA_FASTDDS_STATISTICS_BACKEND_TYPES_ENTITYID_HPP_
 #define _EPROSIMA_FASTDDS_STATISTICS_BACKEND_TYPES_ENTITYID_HPP_
 
+#include <fastdds-statistics-backend/fastdds_statistics_backend_dll.h>
+
+#include <ostream>
 #include <string>
 
 namespace eprosima {
 namespace statistics_backend {
 
-class FASTDDS_STATISTICS_BACKEND_DllAPI EntityId : public std::string
+class EntityId
 {
 public:
 
-    // Inherit std::string constructors
-    using std::string::string;
+    /**
+     * @brief Instantiate an EntityId. The internal value is set to EntityId::invalid
+     */
+    EntityId();
 
     /**
-     * @brief Return the EntityId to refer all entities at once
+     * @brief Get the EntityId to refer all entities at once
      * @return An ID that refers all entities.
      */
-    static EntityId all()
-    {
-        return EntityId(ENTITY_ID_ALL);
-    }
+    FASTDDS_STATISTICS_BACKEND_DllAPI static EntityId all();
 
-private:
-    static constexpr const char * ENTITY_ID_ALL = "*";
+    /**
+     * @brief Get an invalid EntityId
+     * @return An ID that is invalid
+     */
+    FASTDDS_STATISTICS_BACKEND_DllAPI static EntityId invalid();
+
+    /**
+     * @brief Invalidate an EntityId
+     */
+    FASTDDS_STATISTICS_BACKEND_DllAPI void invalidate();
+
+    /**
+     * @brief Check whether an EntityId is valid
+     * @return An true if valid, false otherwise
+     */
+    FASTDDS_STATISTICS_BACKEND_DllAPI bool valid();
+
+    /**
+     * @brief Get the internal value of the EntityId
+     * @return An int64_t with the representing internal value
+     */
+    int64_t value() const;
+
+protected:
+
+    /**
+     * @brief Instantiate an EntityId from an integer.
+     * @param value The value to use as internal value on the EntityId
+     */
+    EntityId(
+            int64_t value);
```

Comment:
We may need the copy constructor:
```
EntityId(const EntityId& id) = default;
```

Reply:
As we discussed offline, we'll added when needed